### PR TITLE
Simplify the greeter and provide more info to new users

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

The greeter provides little information for new users

Issue Number: #1879

### What is the new behavior?

This PR proposes to make the greeter simpler. I think we should not drop users in an empty model. This worked fine in the past, but we like to give users a head start by using a template.

image

When no recent files are present:

image

### CODE REVIEW

**Comment by bertob** [comment]
Looks good to me structurally, but the spacing inside the rows looks a bit odd, the rows are very tall and there's a ton of spacing between title and subtitle. I'd use standard libadwaita row styling here.

File: gaphor/ui/greeter.py [hunk 1]
```diff
@@ -8,25 +8,55 @@
...
+    ModelTemplate(
+        gettext("UML"),
+        gettext("A Unified Modeling Language template/example."),
+        "UML",
+        "UML",
+        "uml.gaphor",
+    ),
+    ModelTemplate(
+        gettext("SysML"),
+        gettext("Systems Modeling template"),
```
Comment (danyeaw) [inline gaphor/ui/greeter.py:41]: I think we should either spell them all out completely or abbreviate them all. SysML is Systems Modeling Language and RAAML is Risk Analysis and Assessment Modeling Language. Why does the UML one say template/example and the others just say template?

**Review by danyeaw — COMMENTED** [review]
I really like where this is heading!

**Comment by amolenaar** [comment]
I updated the GTK4 layout to use `AdwPreferencesGroup`. It looks a bit more compact. I added a little extra padding on the rows.

image

File: gaphor/ui/greeter.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/ui/greeter.py:44]: I quickly dropped some text.

RAAML spelled out is a bit too much text IMHO. Currently the description does not add a lot. 

Something like "Systems Modeling Language template" does not say anything about what you can do with it. I think that the term "template" is also a bit misplaced. You're creating a new model _based on a template_, not the template.

**Comment by bertob** [comment]
Nice, looks much better! A few more minor things:
- I'd simplify the titles a bit and use title case, e.g. "Recently Opened" and "Create New Model"
- I'd move the Open button to the left side of the headerbar and change the lable to "Open Model..."
- It'd be good to already have access to the app's primary menu in this state, so you can e.g. check the version in the about dialog

**Comment by amolenaar** [comment]
@bertob Thanks a lot for your quick feedback.

Something like this (hamburger will follow)?

image

**Comment by bertob** [comment]
Yup, looks good :+1:

**Comment by amolenaar** [comment]
With hamburger menu:

image

**Comment by bertob** [comment]
Hmm, why is there an extra "Close" item in the menu?

I was thinking of something like this: 

```
Open Model...
---
Keyboard Shortcuts
About Gaphor
```

**Comment by amolenaar** [comment]
I had a look at GNOME Builder. It also had a close menu item 😅. There's not really any keyboard shortcuts in this window.

**Comment by bertob** [comment]
Hmmm, generally menus should have at least 3 items :D  

I guess it'd be fine to show all keyboard shortcuts though since at least Ctrl+W, Ctrl+Q, F10, and the like would work here too.

**Comment by bertob** [comment]
With the keyboard entry this is good to go, I think.

**Review by danyeaw — APPROVED** [review]
This is a really great improvement! Thanks @amolenaar for the great implementation and @bertob for the original idea and guidance on the design.

**Pull request merged**